doc/working-with-git-and-github.md

```diff
@@ -153,131 +411,181 @@ To unstash your changes, issue the following command:
-   [GitHub CLI](https://cli.github.com/))
-2. [via SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
-   (hardest method, requires setup, but ultimately very convenient)
+### Sync a local clone
 
-### Add a remote
-
-When you use the operation `git push`, you are pushing your local commit(s) to
-a remote version of your repository.
+Once you or other people have submitted changes and they have been accepted
+and "merged" into the official `main` branch, the local clone you made on your
+machine will become out of date in terms of its view of the code in the main
+branch. (It won't change its concept of what code is in main until you
+explicitly tell it to download or "pull" the latest commits in main.)
 
-First, [check to see what remotes you have](#check-your-remotes).
+This can be a problem -- you may not want to base a new branch on out-of-date
+code, or a PR you may have submitted may have ended up in a conflicted state
+because it was based on an older version of `main`. Here's how you correct the
+situation.
 
-To add a remote version of your repository, issue the following command:
+First, you need to get back onto the main branch in your local working tree.
+If you have any uncommited changes (they would show via
+[`git status`](#display-status)) that you want to save, you should first
+[stash](#stash-your-changes) them. Then execute
 
 ```sh
-git remote add name_of_remote https://someurl.com/somerepo.git
+git checkout main
+git pull
 ```
 
-### Check your remotes
-
-To check to see what your "remotes" (where you'll be pushing to) are, issue
-the command below. (The items in <> brackets are placeholders. You shouldn't
-actually have the <> brackets in the real thing.)
+This will download all of the changes to the standard `main` branch since you
+created your clone or last pulled `main`, and leave your working tree on the
+main branch with all of the new code. Now you if you want to return to the
+branch you were on, you can execute
 
 ```sh
-$ git remote -v
-origin  git@github.com:<org-or-user-name>/<repo-name>.git (fetch)
-origin  git@github.com:<org-or-user-name>/<repo-name>.git (push)
-upstream        git@github.com:<org-or-user-name>/<repo-name>.git (fetch)
-upstream        git@github.com:<org-or-user-name>/<repo-name>.git (push)
+git checkout [branch-name]
 ```
 
-The names "origin" and "upstream" are arbitrary. That being said, they are the
-traditional names. "origin" should be your fork, and "upstream" should be the
-original repository.
+And if you previously stashed any changes, you may want to
+[unstash them](#unstash-your-changes).
+
+### Rebase your branch
 
-### Sync local fork with remote original
+A branch works from a specific commit in its parent branch (usually `main`) --
+its like a logbook of all of the changes you've made from that particular
+snapshot of all of the code. Many times for a PR to succeed, the branch must
```

Comment:
Often, for a PR to succeed.... (I had to read this sentence twice for some reason with the current wording)